### ISSUE TITLE
[Security]: target="_blank" should not be used

### ISSUE THREAD

**Issue by szponek** [haiwen/seahub#1066]
as described at https://mathiasbynens.github.io/rel-noopener/
A quick grep through seahub code revealed ~116 occurrences of links containing target="_blank"

Is it possible to exploit that by i.e. sharing a file with a malicious link or attaching it to a group discussion? Do I need to take any precautions?

Is it necessary to use that in seahub links so widely?

**Comment by shoeper** [comment]
Is there any place it is being used for external links? For internal links there shouldn't be any issues at all.

**Comment by szponek** [comment]
A quick look suggests most of the external links with target="_blank" point to http://seafile.com.

However, I've pasted a link pointing at https://mathiasbynens.be/demo/opener into a group discussion on one of my libraries, clicked on it, a new browser tab opened and when I got back to the original tab it already got redirected to Mathias's 'hacked' site. I assume it would be possible to make that tab do something more malicious (Delete libraries? Share them more widely? Steal a session cookie?) without the need to inject any js code to seafile site.

I admit, this can be done only by another seafile user and same group member and requires target user to click on a link, but it's not an uncommon scenario.
